### ISSUE TITLE
DataFrame.to_csv() produces wrong results when mixed-type columns are specified

### ISSUE THREAD

**Issue by Sereger13** [pydata/pandas#11637]
This used to work OK in version 0.15 but stopped working in versions 0.16+.

```python
import pandas as pd
df = pd.DataFrame({0: ['a', 'b', 'c'],
                   1: ['aa', 'bb', 'cc']})
df['test'] = 'txt'
df.to_csv('/home/temp/df.csv', columns=['test', 0, 1])
```
The following output is produced (values from columns 0 and 1 are missing):
```
,test,0,1
0,txt,,
1,txt,,
2,txt,,
```
Replacing 'test' with an integer value or skipping the columns argument works fine.

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.9.final.0
python-bits: 64
OS: Linux
OS-release: 2.6.18-238.9.1.el5
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US

pandas: 0.17.0
...

**Comment by jreback** [comment]
yeh should probably work, pull-requests to fix are welcome

```
In [11]: df.to_csv()
Out[11]: ',0,1,test\n0,a,aa,txt\n1,b,bb,txt\n2,c,cc,txt\n'

In [12]: df.to_csv(columns=['test',0,1])
Out[12]: ',test,0,1\n0,txt,,\n1,txt,,\n2,txt,,\n'
```

**Comment by jreback** [comment]
closed by #11649
